### PR TITLE
[SPARK-44836][PYTHON] Refactor Arrow Python UDTF

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Refactors Arrow Python UDTF.

### Why are the changes needed?

Arrow Python UDTF is not need to be redefined when creating it. It can be handled in `worker.py`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The existing tests.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @dtenedor @allisonwang-db

File: python/pyspark/worker.py [hunk 1]
```diff
@@ -628,9 +628,36 @@ def verify_result(result):
                 )
                 return result
 
-            return lambda *a, **kw: map(
-                lambda res: (res, arrow_return_type), map(verify_result, f(*a, **kw))
-            )
+            def evaluate(*args: pd.Series, **kwargs: pd.Series):
+                try:
+                    if len(args) == 0 and len(kwargs) == 0:
+                        yield verify_result(pd.DataFrame(f())), arrow_return_type
```
Comment (allisonwang-db) [inline python/pyspark/worker.py:634]: I see we have a try catch statement for the entire evaluate function.  Can we actually only wrap the invocation of `f()`? This is to separate the user's code error from the error thrown by the worker code.

File: python/pyspark/worker.py [hunk 2]
```diff
@@ -628,9 +628,36 @@ def verify_result(result):
...
+                        # Create tuples from the input pandas Series, each tuple
+                        # represents a row across all Series.
+                        keys = list(kwargs.keys())
+                        len_args = len(args)
+                        row_tuples = zip(*args, *[kwargs[key] for key in keys])
+                        for row in row_tuples:
+                            res = f(
+                                *row[:len_args],
+                                **{key: row[len_args + i] for i, key in enumerate(keys)},
+                            )
```
Comment (allisonwang-db) [inline python/pyspark/worker.py:645]: ditto

**Review by allisonwang-db — COMMENTED** [review]
Very nice refactor! Left a comment.

**Comment by ueshin** [comment]
For 3.5: #42522 as this changes Spark Connect client implementation.

**Review by allisonwang-db — APPROVED** [review]
LGTM!

**Review by dtenedor — APPROVED** [review]
Thanks, this should indeed make it easier to work with.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**